src/reanimated2/interpolation.js

```diff
@@ -1,30 +1,117 @@
 import { Extrapolate } from '../derived/interpolate';
 
+function getVal(config) {
+  'worklet';
+
+  const { type, coef, val, ll, rr, x } = config;
+
+  switch (type) {
+    case Extrapolate.IDENTITY:
+      return x;
+    case Extrapolate.CLAMP:
+      if (coef * val < coef * ll) {
+        return ll;
+      }
+      return rr;
+    case Extrapolate.EXTEND:
+    default:
+      return val;
+  }
+}
+
+function isExtrapolate(value) {
+  'worklet';
+
+  return (
+    value === Extrapolate.EXTEND ||
+    value === Extrapolate.CLAMP ||
+    value === Extrapolate.IDENTITY
+  );
+}
+
+function validateType(type) {
+  'worklet';
+
+  if (!type && type !== undefined) {
+    throw new Error(
+      `Reanimated: config object is not valid please provide valid config, for example:
+       interpolate(value, [inputRange], [outputRange], {
+        extrapolateLeft: 'clamp',
+        extrapolateRight: 'extend',
+      }) or interpolate(value, [inputRange], [outputRange], 'clamp')`
+    );
+  }
+
+  if (
+    type?.constructor === Object &&
+    ((Object.keys(type).length === 2 &&
+      !('extrapolateLeft' in type && 'extrapolateRight' in type)) ||
```

Comment:
Nit; `in` operator traverses prototype hierarchy, we should probably use [`type.hasOwnProperty('extrapolateLeft')`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty).

Just a nit, because we expect an object as a dictionary here.

Reply:
Done ✅